source/Octopus.Versioning/Octopus/OctopusVersionMaskParser.cs

```diff
@@ -49,7 +50,16 @@ public OctopusVersionMask Parse(string? version)
         public IVersion ApplyMask(string? mask, IVersion? currentVersion)
         {
             var parsedMask = Parse(mask);
-            if (!parsedMask.IsMask)
+
+            /*
+             * Watch out for this! A mask can return false for IsMask, but still succeed
+             * parsing here and proceed to apply the masking logic. See OctopusVersionMaskParserTests.IsMask
+             * for edge cases where masks fail IsMask but still work as masks.
+             *
+             * See SemanticVersionMask.ApplyMask in the test project for the original logic
+             * that has been replicated here for consistency.
+             */
+            if (!parsedMask.DidParse)
```

Comment:
This is the change that allows masks like `1.2.3-i` to be used, and retains compatibility with the old class.